PlainTasks.py

```diff
@@ -833,3 +842,146 @@ def run(self, edit):
                 msg = msg.replace(o, r)
 
         sublime.set_clipboard(msg)
+
+class PlainTasksArchiveOrgCommand(PlainTasksBase):
+    def runCommand(self, edit):
+        # Archive the curent subtree to our archive file, not just completed tasks.
+        # For now, it's mapped to ctrl-shift-o or super-shift-o
+
+        # TODO: Mark any tasks found as complete, or maybe warn.
+
+        # Get our archive filename
+        archive_filename = self.__createArchiveFilename()
+
+        # Figure out our subtree
+        start_line, end_line = self.__findCurrentSubtree()
+        if (start_line < 0 or end_line < 0):
+            # Errors would have been displayed already
+            return
+
+        # Write our region or our archive file
+        success, region = self.__writeArchive(archive_filename, start_line, end_line)
+
+        # only erase our region if the write was successful
+        if success is True:
```

Comment:
Sweet -- I'll make those changes and re-submit.

On Tue, Feb 24, 2015 at 3:07 PM, Vova Kolobok <notifications@github.com>
wrote:

> In PlainTasks.py
> <https://github.com/aziz/PlainTasks/pull/215#discussion_r25297961>:
>
> > +        # TODO: Mark any tasks found as complete, or maybe warn.
> > +
> > +        # Get our archive filename
> > +        archive_filename = self.__createArchiveFilename()
> > +
> > +        # Figure out our subtree
> > +        start_line, end_line = self.__findCurrentSubtree()
> > +        if (start_line < 0 or end_line < 0):
> > +            # Errors would have been displayed already
> > +            return
> > +
> > +        # Write our region or our archive file
> > +        success, region = self.__writeArchive(archive_filename, start_line, end_line)
> > +
> > +        # only erase our region if the write was successful
> > +        if success is True:
>
> if success:
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aziz/PlainTasks/pull/215/files#r25297961>.
>